addons/spreadsheet/static/src/pivot/pivot_table.js

```diff
@@ -171,8 +180,80 @@ export class SpreadsheetPivotTable {
         return this._rows[index];
     }
 
-    getRowTitle() {
-        return this._rowTitle;
+    /**
+     * @returns {PivotCell[][]}
+     */
+    getPivotCells() {
+        const pivotHeight = this.getColHeadersHeight() + this.getRowHeight();
+        const pivotWidth = this.getColWidth() + 1;
+
+        const domainArray = [];
+        for (let col = 0; col < pivotWidth; col++) {
+            domainArray.push([]);
+            for (let row = 0; row < pivotHeight; row++) {
+                domainArray[col].push(this._getPivotCell(col, row));
+            }
+        }
+        return domainArray;
+    }
+
+    /**
+     * @returns {PivotCell}
+     */
+    _getPivotCell(col, row) {
+        const colHeadersHeight = this.getColHeadersHeight();
+        if (col === 0 && row === colHeadersHeight - 1) {
+            return { content: this._rowTitle, isHeader: true, style: HEADER_STYLE };
+        } else if (row <= colHeadersHeight - 1) {
+            const domain = this._getColHeaderDomain(col, row);
+            const style = row === colHeadersHeight - 1 ? MEASURE_STYLE : TOP_LEVEL_STYLE;
+            return { domain, isHeader: true, style };
+        } else if (col === 0) {
+            const rowIndex = row - colHeadersHeight;
+            const domain = this._getRowDomain(rowIndex);
+            const indent = this._rows[rowIndex].indent;
+            //TODO DISCUSS: isn't undefined style on header a ugly ?
```

Comment:
Personally, I don't really like it but turns out it will end up updating history with the value undefined whether you provide { style: undefined} or don't provide the key at all...

Reply:
It doesn't choque me plus que ça

Reply:
let's keep it as it is then